### PR TITLE
TD-4429 Additional optional competencies checkbox on the supervisor sign off screen

### PR DESCRIPTION
**Pull request by sherif-olaboye**
### JIRA link
https://hee-tis.atlassian.net/browse/TD-4429

### Description
I resolved unboxing a possible null value warning in the AddSuperviseDelegate and InsertCandidateAssessmentSupervisor method
I added NumberOfSelfAssessedOptionalCompetencies count to the get and post SignOffProfileAssessment  action method in the Supervisor.cs
 added MinimumOptionalCompetencies and DelegateUserID field to GetCandidateAssessmentSupervisor query
I added NumberOfSelfAssessedOptionalCompetencies  and OptionalCompetenciesChecked  property in the RequestSignOffViewModel.cs
I introduce a new checkbox, label, and validation in SignOffProfileAssessment.cshtml
### Screenshots
image
image
image
image

-----
### Developer checks
(Leave tasks unticked if they haven't been appropriate for your ticket.)

I have:
- [x] Run the formatter and made sure there are no IDE errors (see [info on Text Editor settings](https://hee-tis.atlassian.net/wiki/spaces/TP/pages/3546185813/DLS+Dev+Process) to avoid whitespace changes)
- [ ] Written tests for the changes (accessibility tests, unit tests for controller, data services, services, view models, etc)
- [x] Manually tested my work with and without JavaScript
- [x] Tested any Views or partials created or changed with [Wave Chrome plugin](https://chrome.google.com/webstore/detail/wave-evaluation-tool/jbbplnpkjmmeebjpijfedlgcdilocofh/related) and addressed any valid accessibility issues
- [ ] Updated/added documentation in [Confluence](https://hee-tis.atlassian.net/wiki/spaces/TP/pages/3546939432/DLS+Code) and/or [GitHub Readme](https://github.com/TechnologyEnhancedLearning/DLSV2/blob/master/README.md). List of documentation links added/changed:
  - [doc_1_here](link_1_here)
- [ ] Updated my Jira ticket with information about other parts of the system that were touched as part of the MR and have to be sanity tested to ensure nothing’s broken
- [x] Scanned over my pull request in GitHub and addressed any warnings from the GitHub Build and Test checks.

### CODE REVIEW

File: DigitalLearningSolutions.Web/Views/Supervisor/SignOffProfileAssessment.cshtml [hunk 1]
```diff
@@ -201,7 +204,18 @@
             <span asp-validation-for="SignedOff" class="text-danger"></span>
           </div>
         </nhs-form-group>
-      </fieldset>
+                @if (Model.NumberOfSelfAssessedOptionalCompetencies >= Model.SelfAssessmentResultSummary.MinimumOptionalCompetencies)
```
Comment (rshrirohit) [inline DigitalLearningSolutions.Web/Views/Supervisor/SignOffProfileAssessment.cshtml:207]: As discussed over a call, let's keep the original scope of the ticket and avoid this unnecessary condition.

**Review by rshrirohit — CHANGES_REQUESTED** [review]
Please see the inline comment. Remove other code changes done in this PR in accordance with the change asked in in-line comment on the SignOffProfileAssessment.cshtml file.

File: DigitalLearningSolutions.Web/Views/Supervisor/SignOffProfileAssessment.cshtml (hunk 1 above)
Comment (sherif-olaboye) [reply DigitalLearningSolutions.Web/Views/Supervisor/SignOffProfileAssessment.cshtml]: Done

File: DigitalLearningSolutions.Data/DataServices/SupervisorDataService.cs [hunk 2]
```diff
@@ -1067,7 +1067,7 @@ SelfAssessmentStructure AS sas1 INNER JOIN
              (ca1.ID = ca.ID) AND (caoc1.IncludedInSelfAssessment = 1) AND (NOT (sar1.Result IS NULL)) AND (sasrv.SignedOff = 1) AND (caqrr1.LevelRAG = 3) OR
              (ca1.ID = ca.ID) AND (sas1.Optional = 0) AND (NOT (sar1.SupportingComments IS NULL)) AND (sasrv.SignedOff = 1) AND (caqrr1.LevelRAG = 3) OR
              (ca1.ID = ca.ID) AND (caoc1.IncludedInSelfAssessment = 1) AND (NOT (sar1.SupportingComments IS NULL)) AND (sasrv.SignedOff = 1) AND (caqrr1.LevelRAG = 3)) AS MeetingCount,
-              sa.SignOffSupervisorStatement
+              sa.SignOffSupervisorStatement,ca.DelegateUserID
```
Comment (rshrirohit) [inline DigitalLearningSolutions.Data/DataServices/SupervisorDataService.cs:1070]: why do we add DelegateUserID here?

**Pull request merged**